plugins/module_utils/network/sonic/config/qos_interfaces/qos_interfaces.py

```diff
@@ -403,144 +386,91 @@ def get_delete_qos_interfaces_requests(self, commands, have, is_delete_all):
                 pfc_priority_queue = qos_maps.get('pfc_priority_queue')
                 pfc_priority_pg = qos_maps.get('pfc_priority_pg')
 
-                cfg_dscp_fwd_group = cfg_qos_maps.get('dscp_fwd_group')
-                cfg_dot1p_fwd_group = cfg_qos_maps.get('dot1p_fwd_group')
-                cfg_fwd_group_dscp = cfg_qos_maps.get('fwd_group_dscp')
-                cfg_fwd_group_dot1p = cfg_qos_maps.get('fwd_group_dot1p')
-                cfg_fwd_group_queue = cfg_qos_maps.get('fwd_group_queue')
-                cfg_fwd_group_pg = cfg_qos_maps.get('fwd_group_pg')
-                cfg_pfc_priority_queue = cfg_qos_maps.get('pfc_priority_queue')
-                cfg_pfc_priority_pg = cfg_qos_maps.get('pfc_priority_pg')
-
-                if dscp_fwd_group and dscp_fwd_group == cfg_dscp_fwd_group:
+                if dscp_fwd_group:
                     requests.append(self.get_delete_map_request(name, 'dscp-to-forwarding-group'))
-                    maps_dict['dscp_fwd_group'] = dscp_fwd_group
-                if dot1p_fwd_group and dot1p_fwd_group == cfg_dot1p_fwd_group:
+                if dot1p_fwd_group:
                     requests.append(self.get_delete_map_request(name, 'dot1p-to-forwarding-group'))
-                    maps_dict['dot1p_fwd_group'] = dot1p_fwd_group
-                if fwd_group_dscp and fwd_group_dscp == cfg_fwd_group_dscp:
+                if fwd_group_dscp:
                     requests.append(self.get_delete_map_request(name, 'forwarding-group-to-dscp'))
-                    maps_dict['fwd_group_dscp'] = fwd_group_dscp
-                if fwd_group_dot1p and fwd_group_dot1p == cfg_fwd_group_dot1p:
+                if fwd_group_dot1p:
                     requests.append(self.get_delete_map_request(name, 'forwarding-group-to-dot1p'))
-                    maps_dict['fwd_group_dot1p'] = fwd_group_dot1p
-                if fwd_group_queue and fwd_group_queue == cfg_fwd_group_queue:
+                if fwd_group_queue:
                     requests.append(self.get_delete_map_request(name, 'forwarding-group-to-queue'))
-                    maps_dict['fwd_group_queue'] = fwd_group_queue
-                if fwd_group_pg and fwd_group_pg == cfg_fwd_group_pg:
+                if fwd_group_pg:
                     requests.append(self.get_delete_map_request(name, 'forwarding-group-to-priority-group'))
```

Comment:
The repetition of these similar blocks and the instructions in lines 380-387 could be reduced to a dictionary or list of dictionaries mapping argspec names to OC names, plus a 'for' loop with two instructions iterating over the dictionary or list entries.

For example:

map_types_to_oc = {
  'dscp_fwd_group': 'dscp-to-forwarding-group',
  'dot1p_fwd_group': 'dot1p-to-forwarding-group',
                                   *
                                   *
                                   *
}

for map_type in map_types_to_oc:
  if qos_maps.get(map_type):
    requests.append(self.get_delete_map_request(name, map_types_to_oc[map_type])